lib/ruby_lsp/executor.rb

```diff
@@ -275,7 +275,12 @@ def on_type_formatting(uri, position, character)
       ).returns(T::Array[Interface::DocumentHighlight])
     end
     def document_highlight(uri, position)
-      Requests::DocumentHighlight.new(@store.get(uri), position).run
+      document = @store.get(uri)
```

Comment:
We can return early if the document has syntax errors

Reply:
Since `executor.rb` has `emitter.visit(document.tree) if document.parsed?`, do we need to be concerned with that here?

Reply:
Where? It seems we're visiting the tree a few lines down and we're not checking if the document is parsed.

Reply:
Ah, ignore, I was looking at the wrong section.

Reply:
Updated 👍